src/Health/AbstractHealth.php

```diff
@@ -0,0 +1,45 @@
+<?php
+declare(strict_types=1);
+
+namespace EoneoPay\Externals\Health;
+
+use EoneoPay\Externals\Health\Interfaces\HealthInterface;
+
+abstract class AbstractHealth implements HealthInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    abstract public function getChecks(): array;
```

Comment:
The health checks should be injected in the constructor.

```
final class Health
{
  private $checks;

  public function __construct(array $checks) { ... }
}
```

You then define a service provider in externals that will gather all checks from the container (https://github.com/eonx-com/requesthandlers/blob/development/src/Bridge/Laravel/Providers/ParamConverterProvider.php#L168-L176)


Using this approach means that we do not need to extend Health in any project, just need to simply use it. We are able to tag our own services in the applications service providers and they will be automatically picked up.